lib/experimental/fonts/font-library/font-library.php

```diff
@@ -187,3 +187,60 @@ function wp_get_font_dir( $defaults = array() ) {
 		return apply_filters( 'font_dir', $defaults );
 	}
 }
+
+// @core-merge: Filters should go in `src/wp-includes/default-filters.php`,
+// functions in a general file for font library.
+if ( ! function_exists( '_wp_deleted_font_family' ) ) {
+	/**
+	 * Deletes child font faces when a font family is deleted.
+	 *
+	 * @access private
+	 * @since 6.5.0
+	 *
+	 * @param int     $post_id Post ID.
+	 * @param WP_Post $post    Post object.
+	 * @return void
+	 */
+	function _wp_delete_font_family( $post_id, $post ) {
+		if ( 'wp_font_family' !== $post->post_type ) {
+			return;
+		}
+
+		$font_faces = get_children(
+			array(
+				'post_parent' => $post_id,
+				'post_type'   => 'wp_font_face',
+			)
+		);
+
+		foreach ( $font_faces as $font_face ) {
+			wp_delete_post( $font_face->ID, true );
+		}
+	}
+	add_action( 'deleted_post', '_wp_delete_font_family', 10, 2 );
```

Comment:
Should this be `delete` rather than `deleted` too?

```suggestion
	add_action( 'delete_post', '_wp_delete_font_family', 10, 2 );
```

Reply:
The hook is intentionally `deleted_post`, which happens after the font family has been deleted, rather than before. That way we're waiting to make sure it gets deleted, first, before deleting child font faces.

With the font face, we need to run the hook earlier, because the post meta needs to still exist to find the files associated with the font face.

Reply:
Ah that makes sense 👍